src/java/org/apache/cassandra/repair/autorepair/RepairTokenRangeSplitter.java

```diff
@@ -224,6 +215,16 @@ public Iterator<KeyspaceRepairAssignments> getRepairAssignments(boolean primaryR
         return new BytesBasedRepairAssignmentIterator(primaryRangeOnly, repairPlans);
     }
 
+    @Override
+    public long adjustTotalBytesToRepair(long totalBytesToRepair)
+    {
+        if (totalBytesToRepair > maxBytesPerSchedule.toBytes())
+        {
+            return maxBytesPerSchedule.toBytes();
+        }
+        return totalBytesToRepair;
```

Comment:
Can be simplified using `Math.min()`, e.g.:

```suggestion
        return Math.min(totalBytesToRepair, maxBytesPerSchedule.toBytes());
```

Reply:
One concern here is that `maxBytesPerSchedule` is an upper bound, but not actually how many bytes may be repaired in a schedule.  I think if we complete our schedule plan, maybe we should just adjust the totalBytesToRepairMetric.

I suppose we may have the same problem with `totalKeyspaceRepairPlansToRepair` as we may not actually repair all keyspaces.